trieste/acquisition/function.py

```diff
@@ -617,6 +617,118 @@ def ehvi_based_on_partitioned_cell(
     return acquisition
 
 
+class BatchMonteCarloExpectedHypervolumeImprovement(SingleModelAcquisitionBuilder):
+    """
+    Builder for the batch expected hypervolume improvement acquisition function.
+    The implementation of the acquisition function largely
+    follows :cite:`daulton2020differentiable`
+    """
+
+    def __init__(self, sample_size: int = 512, *, jitter: float = DEFAULTS.JITTER):
+        """
+        :param sample_size: The number of samples for each batch of points.
+        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
+            the covariance matrix.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive, or
+            ``jitter`` is negative.
+        """
+        tf.debugging.assert_positive(sample_size)
+        tf.debugging.assert_greater_equal(jitter, 0.0)
+
+        super().__init__()
+
+        self._sample_size = sample_size
+        self._jitter = jitter
+        self.q = -1
+
+    def __repr__(self) -> str:
+        """"""
+        return (
+            f"BatchMonteCarloHypervolumeExpectedImprovement({self._sample_size!r},"
+            f" jitter={self._jitter!r})"
+        )
+
+    def _cache_q_subset_indices(self, q: int) -> None:
+        """
+        Cache indices containing all subsets of `q`.
+
+        :param q: batch size
+        """
+        if q != self.q:  # regenerate subset indices
+            indices = list(range(q))
+            self.q_subset_indices = {
+                f"q_choose_{i}": tf.constant(list(combinations(indices, i)))
```

Comment:
It's a bit weird to have a string index like this when you could just use `i`? Also, you don't need to listify indices, as combinations accepts an iterable.

```suggestion
                i: tf.constant(list(combinations(range(q), i)))
```

Reply:
Sorry for the late response and thanks for the comments. Just thinking maybe it helps to clarify why we earlier have the `q_choose_{}` here. 

The string is more for interpretability. For instance, when `i=2`, means we are doing inclusion-exclusion with the case we are only choose 2 point from `q` point, ofc it can be removed or replaced by comment if that is a more proper way :)